### ISSUE TITLE
No Incomplete Cholesky Factorization in Ifpack2

### ISSUE THREAD

**Issue by davoodansari** [trilinos/Trilinos#4521]
Migrating from Epetra to Tpetra, unfortunately Ifpack2 does not seem to have an ICT preconditioner available. Symmetric operators occur in many problems and the availability of an ICT (just like legacy Ifpack) is probably more than a nice to have. Any comments about long term plans or advice on short term solution or alternatives is highly appreciated. 

Thanks

@trilinos/<teamName>

## Your Environment

- **Relevant repo SHA1s:**  
- **Relevant configure flags or configure script:**  
- **Operating system and version:**  
- **Compiler and TPL versions:**  

## Related Issues

* Blocks 
* Is blocked by 
* Follows 
* Precedes 
* Related to 
* Part of 
* Composed of

**Comment by mayrmt** [comment]
@trilinos/ifpack2

**Comment by mhoemmen** [comment]
We welcome a pull request that ports Ifpack's ICT to Ifpack2, as long as it comes with tests.

**Comment by jhux2** [comment]
@srajama1

**Comment by srajama1** [comment]
@davoodansari We will try and get this ported to Ifpack2. For some problems you might be able to get away with FAST_IC and/or FAST_ILDL. You would need to enable ShyLU for those. Note that they are not an exact equivalent of the ICT as they are basically IC(k) style approaches. In addition, they are also iterative variants of IC(k).

**Comment by davoodansari** [comment]
Thank you @srajama1 That would be great! Is there a tentative timeline based on which we might be able to plan things out?

**Comment by srajama1** [comment]
Honestly, it is hard to give a time line for these as internally current funded projects don't need it. In addition to your request, I see one use case in a funded project where this might be needed. However, they are not actively asking for it yet. This is a need, but prioritizing is dependent on lot of other factors.  I hope you understand.

**Comment by jhux2** [comment]
> Honestly, it is hard to give a time line for these as internally current funded projects don't need it. In addition to your request, I see one use case in a funded project where this might be needed. However, they are not actively asking for it yet. This is a need, but prioritizing is dependent on lot of other factors. I hope you understand.

@davoodansari Might you have a patch that you can submit?  This would help speed up the process.

**Comment by davoodansari** [comment]
@jhux2 I wish we had one. We are working with very stringent resources and for now. The best short term solution that came to my mind was to use the old Ifpack until things change. On that front I actually submitted an email to the help thread to see if any one  has experience or examples they could share showing conversions from between Ifpack and Ifpack2 (and of course Tpetra and Eptera).

**Comment by davoodansari** [comment]
@srajama1 I fully understand. I am grateful that you guys are at least planing to work on it.

**Comment by mhoemmen** [comment]
@davoodansari wrote:
> On that front I actually submitted an email to the help thread to see if any one has experience or examples they could share showing conversions from between Ifpack and Ifpack2 (and of course Tpetra and Epetra).

Check out the Tpetra tutorial here: https://github.com/trilinos/Trilinos/tree/master/packages/tpetra/core/example (look for directories starting with "Lesson0").

**Comment by davoodansari** [comment]
> @davoodansari wrote:
> 
> > On that front I actually submitted an email to the help thread to see if any one has experience or examples they could share showing conversions from between Ifpack and Ifpack2 (and of course Tpetra and Epetra).
> 
> Check out the Tpetra tutorial here: https://github.com/trilinos/Trilinos/tree/master/packages/tpetra/core/example (look for directories starting with "Lesson0").

@mhoemmen Thank you. I may have misrepresented my problem. Using those tutorials I am able to use Tpetra and other libraries like Tpetra/Ifpack2. What I really need is a way to bridge between Ifpack2 and the older Ifpack/Epetra (for example) because currently there are functionalities in Ifpack that are not yet available in Ifpack2. I am not sure though if that is actually possible.
